File: JavaScript-Node/programs/linear-number-interpolation/interpolation.js
function interpolate(x, y, precission = 1) {
    precission = precission + 1;
    var toreturn = [];
    var smallnumber = x;
    var diference = y - x;

    if(x > y) {
        smallnumber = y;
        diference = x - y;
    }

    var diferenceByPrecission = diference / precission;

    for(var i = 1; i < precission; i++) {
        toreturn[i-1] = smallnumber + (diferenceByPrecission * i);
    }

    return toreturn;
}

module.exports = {
    interpolate
};

File: C/programs/linear-number-interpolation/interpolation.c
#include <stdlib.h>

#include "interpolation.h"

struct ArrayObject *interpolate(float x, float y, int precission)
{
    float *arr;
    arr = (float*) calloc(precission, sizeof(float));

    precission = precission + 1;
    float smallnumber = x;
    float diference = y - x;

    if(x > y) {
        smallnumber = y;
        diference = x - y;
    }

    float diferenceByPrecission = diference / precission;

    for(int i = 1; i < precission; i++) {
        arr[i-1] = smallnumber + (diferenceByPrecission * i);
    }

    static struct ArrayObject toreturn = {0, 0};

    toreturn.len = precission-1;
    toreturn.Array = arr;
    
    return &toreturn;
}